packages/account/src/Components/api-token/__tests__/api-token.spec.js

```diff
@@ -0,0 +1,340 @@
+
+        const checkboxes = await screen.findAllByRole('checkbox');
+        const read_checkbox = checkboxes.find(card => card.name === 'read');
+        const create_btn = await screen.findByRole('button');
+        const token_name_input = await screen.findByLabelText('Token name');
+
+        expect(checkboxes.length).toBe(5);
+        expect(read_checkbox.checked).toBeFalsy();
+        expect(create_btn).toBeDisabled();
+        expect(token_name_input.value).toBe('');
+
+        fireEvent.click(read_checkbox);
+        expect(read_checkbox.checked).toBeTruthy();
+
+        fireEvent.change(token_name_input, { target: { value: '@#$' } });
+
+        expect(await screen.findByText('Only letters, numbers, and underscores are allowed.')).toBeInTheDocument();
+
+        fireEvent.change(token_name_input, { target: { value: 'N' } });
+
+        expect(await screen.findByText(/length of token name must be between/i)).toBeInTheDocument();
+
+        fireEvent.change(token_name_input, { target: { value: 'New test extra long name for erorr' } });
+
+        expect(await screen.findByText(/maximum/i)).toBeInTheDocument();
+
+        fireEvent.change(token_name_input, { target: { value: 'New token name' } });
+
+        expect(token_name_input.value).toBe('New token name');
+
+        await waitFor(() => {
+            expect(create_btn).toBeEnabled();
+        });
+
+        fireEvent.click(create_btn);
+
+        const updated_token_name_input = await screen.findByLabelText('Token name');
+        expect(updated_token_name_input.value).toBe('');
+
+        const createToken = mock_props.ws.apiToken;
+        expect(await createToken).toHaveBeenCalledTimes(1);
+    });
+
+    it('should render created tokens and trigger delete', async () => {
+        jest.useFakeTimers();
+
+        getPropertyValue.mockReturnValue([
+            {
+                display_name: 'First test token',
+                last_used: '',
+                scopes: ['Read', 'Trade'],
+                token: 'GBjsG2kM1uxtJtk',
+                valid_for_ip: '',
+            },
+            {
+                display_name: 'Second test token',
+                last_used: '',
+                scopes: ['Read', 'Payments', 'rade'],
+                token: 'GHjaD2f4gDg5gSE',
+                valid_for_ip: '',
+            },
+        ]);
+
+        render(<ApiToken {...mock_props} />);
+
+        expect(await screen.findByText('Name')).toBeInTheDocument();
+        expect(await screen.findByText('Token')).toBeInTheDocument();
+        expect(await screen.findByText('Scopes')).toBeInTheDocument();
+        expect(await screen.findByText('Last used')).toBeInTheDocument();
+        expect(await screen.findByText('Action')).toBeInTheDocument();
+        expect(await screen.findByText('First test token')).toBeInTheDocument();
+        expect(await screen.findByText('Second test token')).toBeInTheDocument();
+
+        const delete_btns_1 = await screen.findAllByText('Delete');
```

Comment:
I would also like to suggest that you consider finding buttons by their role as buttons 🙂  , e.g.:
```suggestion
        const delete_btns_1 = await screen.findAllByRole('button', { name: /delete/i });
```
* `'Delete'` (instead of `/delete/i` ) will also do though )

Reply:
it is great note, I think I had some confusion with passing the name, because I didn’t manage it from the first time. Now it works, thanks :+1: